### ISSUE TITLE
zen_workaround.py PermissionError (Ryzen 3700U)

### ISSUE THREAD

**Issue by bcampbell** [rr-debugger/rr#2933]
I'm having trouble getting the zen_workaround.py script working on my Ryzen 3700U-based laptop.

`lscpu` reports cpu model is "AMD Ryzen 7 3700U with Radeon Vega Mobile Gfx"

I'm using rr 5.4 as packaged by Debian (I've diffed the workaround script and it's the same as in the github repo, other than a link referencing the Zen page on the mozilla wiki rather than the github wiki).
```
$ sudo /usr/share/doc/rr/examples/zen_workaround.py 
Traceback (most recent call last):
  File "/usr/share/doc/rr/examples/zen_workaround.py", line 46, in <module>
    os.write(msr, struct.pack('<q', val))
PermissionError: [Errno 1] Operation not permitted
```

So it's opening `/dev/cpu/7/msr` and seeking, but the actual write fails...
Anything obvious I'm missing?

**Comment by khuey** [comment]
What kernel is this?

**Comment by bcampbell** [comment]
oops, should have mentioned that:
```
$ uname -r
5.10.0-1-amd64
```

**Comment by surr1234** [comment]
Same problem on kernel 5.13.12-200.fc34.x86_64, rr version 5.4.0

**Comment by theIDinside** [comment]
I've solved this issue;

The problem this boils down to is that secure is turned on. From what I can gather, this makes it so the Linux Kernel entirely prevents writing to MSR. If you disable secure boot, the zen_workaround.py script will work.

**Comment by GitMensch** [comment]
Th current script explicit checks the permission error and tries to produce nice user-readable errors.
Can anyone that had the issue recheck if that works? I think this issue can be closed (also: @theIDinside the wiki entry now mentions both `br_inst_retired.conditional ` and `br_inst_retired.cond`, I guess you have the later?)

**Comment by theIDinside** [comment]
@GitMensch yes; I forgot to tag this specific issue when I submitted the changes to the script that checks for UEFI secure boot. I've only had the possibility to test it on my machine and it works (famous last words). But from what I can gather, a lot of linux'es use the same setup, possibly all of them (UEFI variables being in the same place, which is where the script reads the settings from) - and as far as UEFI being the culprit for it not working, discussions around it can be found for instance [here](https://bugzilla.redhat.com/show_bug.cgi?id=1673184)
